core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -744,17 +744,18 @@ class DynamicThreadPool(server: KafkaBroker) extends BrokerReconfigurable {
   }
 }
 
-class DynamicMetricsReporters(brokerId: Int, server: KafkaBroker) extends Reconfigurable {
+class DynamicMetricsReporters(brokerId: Int, config: KafkaConfig, metrics: Metrics, clusterId: String) extends Reconfigurable {
```

Comment:
Could potentially leave this signature alone, but it would probably have to change to this when we implement https://issues.apache.org/jira/browse/KAFKA-14057 anyway, so it perhaps doesn't hurt to change it now?